src/V3EmitCHeaders.cpp

```diff
@@ -129,28 +128,16 @@ class EmitCHeader final : public EmitCConstInit {
                 if (varp->isParam() && (varp->isUsedParam() || varp->isSigPublic())) {
                     decorateFirst(first, "\n// PARAMETERS\n");
                     UASSERT_OBJ(varp->valuep(), nodep, "No init for a param?");
-                    // These should be static const values, however older MSVC++ did't
-                    // support them; should be ok now under C++11, need to refactor.
-                    if (varp->isWide()) {  // Unsupported for output
-                        putsDecoration("// enum WData " + varp->nameProtect() + "  //wide");
-                    } else if (varp->isString()) {
-                        puts("static const std::string " + protect("var_" + varp->name()) + ";\n");
-                    } else if (!VN_IS(varp->valuep(), Const)) {  // Unsupported for output
-                        // putsDecoration("// enum ..... "+varp->nameProtect()
-                        //               +"not simple value, see variable above instead");
-                    } else if (VN_IS(varp->dtypep(), BasicDType)
-                               && VN_CAST(varp->dtypep(), BasicDType)
-                                      ->isOpaque()) {  // Can't put out e.g. doubles
-                    } else {
-                        // enum
-                        puts(varp->isQuad() ? "enum _QData" : "enum _IData");
-                        puts("" + varp->nameProtect() + " { " + varp->nameProtect() + " = ");
-                        iterateAndNextNull(varp->valuep());
-                        puts("};\n");
-                        // var
-                        puts(varp->isQuad() ? "static const QData " : "static const IData ");
-                        puts(protect("var_" + varp->name()) + ";\n");
+                    // std::string is not a literal type, so cannot be constexpr
+                    const bool containsString = containsStringDtype(varp->dtypep());
```

Comment:
I think instead you should look for the opposite, only do this if it's a basic type, and a POD basic type. so you'd call AstBasicDType::isPod which is a new function that calls AstBasicDTypeKwd::isPod.  Similar below.

I didn't understand the associative array check, I think that will fall out of above change.

Reply:
Yup, that's a lot better, thank you.